resources/config.rb

```diff
@@ -57,9 +57,11 @@
     begin
       tempfile.write(TOML::Generator.new(new_resource.config).body)
       tempfile.close
-      shell_out_compact!(['hab', 'config', 'apply', opts,
-                          new_resource.service_group, incarnation,
-                          tempfile.path])
+      args = ['hab', 'config', 'apply', opts,
+              new_resource.service_group, incarnation,
+              tempfile.path].flatten.compact.join(' ')
+
+      shell_out(args)
```

Comment:
Please preserve use of `shell_out_compact!` - via Chef source where this method is defined:

> `# generally its best to use shell_out_compact! in code and setup expectations on shell_out! in tests`

This method is used with the `clean_array` method, and was intentionally done when this was all originally written.

Reply:
Similar to my above comment `clean_array` is also in the call stack for the `shell_out_compact!` method.